From 6971971729dbbea1db8c1006c99b1035f30214a2 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gentoo.org>
Date: Wed, 21 Mar 2012 18:24:26 -0400
Subject: [PATCH] Use wxversion.select('2.8') to support systems that have
 wxpython-2.9 installed

Pyfa-1.x supports only wxpython-2.8, but multiple wxpython versions can
be installed in parallel, and by default, 'import wx' pulls in the
latest one.

To support systems that have both wxpython-2.8 and 2.9 installed, pyfa
needs to call wxversion.select.

Note that wxversion.select can potentially fail with several exceptions
internal to the wxversion module, so for simplicity, just check for all
of them with an 'except:'.

http://trac.evefit.org/ticket/475
---
 pyfa.py |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyfa.py b/pyfa.py
index aedb718..6992aaf 100755
--- a/pyfa.py
+++ b/pyfa.py
@@ -29,8 +29,8 @@ if not hasattr(sys, 'frozen'):
 
     try:
         import wxversion
-        wxversion.ensureMinimal('2.8')
-    except ImportError:
+        wxversion.select('2.8')
+    except:
         print "Cannot find wxPython or the installed wxPython version doesn't meet the min. requirements.\nYou can download wxPython (2.8) from http://www.wxpython.org/"
         sys.exit(1)
 
-- 
1.7.8.5

